Library/Homebrew/resource.rb

```diff
@@ -145,7 +145,10 @@ def url(val = nil, specs = {})
   end
 
   def version(val = nil)
-    @version ||= detect_version(val)
+    @version ||= begin
+      version = detect_version(val)
+      version.is_a?(Version::NullVersion) ? nil : version
```

Comment:
I haven't tracked down the nil check yet, but this has to do with versions being propagated from the parent. If a resource's version isn't parseable from the URL, it inherits the version of its parent resource (e.g., the formula). If the version is assigned as a `NullVersion` instead of `nil`, then it fails to inherit the parent's version.